z/btree.go

```diff
@@ -377,7 +404,7 @@ func (t *Tree) split(pid uint64) node {
 	nn := t.newNode(n.bits())
 	// Re-read n as the underlying buffer for tree might have changed during newNode.
 	n = t.node(pid)
-	rightHalf := n[keyOffset(maxKeys/2):keyOffset(maxKeys)]
+	rightHalf := n[keyOffset(maxKeys/2):keyOffset(maxKeys-1)]
```

Comment:
this is where I start getting uncomfortable with this PR. I will have to better study this btree impl.